code/game/objects/structures/cannons/mounted_guns/mounted_gun.dm

```diff
@@ -186,3 +227,133 @@
+		playsound(src, 'sound/effects/magic/clockwork/fellowship_armory.ogg', 50, FALSE, 5)
+		loaded_gun = TRUE // Make sure it registers theres ammo in there, so it can fire.
+		if(shots_in_gun >= max_shots_per_fire)
+			shots_in_gun = max_shots_per_fire // in case of somehow firing only some of a guns shots, and reloading, you still cant get above the maximum ammo size.
+			fully_loaded_gun = TRUE //So you cant load extra.
+		return
+
+	else
+		user.log_message("fired a cannon", LOG_ATTACK)
+		log_game("[key_name(user)] fired a cannon in [AREACOORD(src)]")
+		addtimer(CALLBACK(src, PROC_REF(fire)), fire_delay) //uses fire proc as shown below to shoot the gun
+		return
+
+/obj/structure/mounted_gun/ratvarian_repeater/fire()
+	if (!loaded_gun)
+		balloon_alert_to_viewers("clockwork mechanism not wound!","",2)
+		return
+	is_firing = TRUE
+	for(var/times_fired = 1, times_fired <= shots_in_gun, times_fired++) //The normal DM for loop structure since the times it has fired is changing in the loop itself.
+		if(loaded_gun)
+			icon_state = icon_state_fire
+			if (times_fired < shots_in_gun)
+				playsound(src, fire_sound, 50, FALSE, 5)
+			else
+				playsound(src, last_fire_sound, 50, TRUE, 5) //for the empty fire sound
+			var/obj/projectile/fired_projectile = new projectile_type(get_turf(src))
+			fired_projectile.firer = src
+			fired_projectile.fired_from = src
+			fired_projectile.fire(dir2angle(dir))
+		if(times_fired % 2 != 1)//Burst Fire.
+			sleep(shot_delay)
+		sleep(shot_delay)
+	loaded_gun = FALSE
+	shots_in_gun = 0
+	fully_loaded_gun = FALSE
+	is_firing = FALSE
+	icon_state = icon_state_base
+
+/obj/structure/mounted_gun/large_ballista
+
+	name = "Improvised Ballista"
+	desc = "''Engineers like to solve problems. If there are no problems handily available, they will create their own problems.''"
+	icon_state = "Improvised_Ballista"
+	icon_state_base = "Improvised_Ballista"
+	icon_state_loaded = "Improvised_Ballista_Loaded"
+	icon_state_fire = "Improvised_Ballista"
+	loading_message = "You finish loading the ballista with a spear."
+	anchored = FALSE
+	anchorable_gun = TRUE
+	uses_ammo = TRUE
+	has_loading_delay = TRUE
+	load_delay = 60
+	max_shots_per_fire = 1
+	shots_per_load = 1
+	shots_in_gun = 1
+	fire_sound = 'sound/items/xbow_lock.ogg'
+	last_fire_sound = 'sound/items/xbow_lock.ogg'
+	has_alt_ammo = TRUE
+	ammo_type = /obj/item/spear
+	projectile_type = /obj/projectile/bullet/Large_Ballista_Spear
+	alt_ammo_type = /obj/item/spear/dragonator
+	alt_projectile_type = /obj/projectile/bullet/Large_Ballista_Spear_Dragonator
+	loaded_gun = FALSE
+	fully_loaded_gun = FALSE
+	fire_delay = 1
+	shot_delay = 1
+	firing_shakes_camera = FALSE
+
+/obj/structure/mounted_gun/large_ballista/attackby(obj/item/ammo_casing/used_item, mob/user, params) //again its single shot so its kinda weird.
+	if(is_firing)
+		balloon_alert(user, "the gun is in the middle of firing!")
+		return
+
+	if(istype(used_item, ammo_type || alt_ammo_type))
```

Comment:
```suggestion
	if(!istype(used_item, ammo_type))
		return
```
Invert the check and return here, then you can unindent everything else
also I don't think passing two types into istype is valid syntax? it is probably only checking for `ammo_type` and would only check for `alt_ammo_type` if `ammo_type` was false or undefined

...but because your alt ammo type is a subtype of your ammo it works anyway